### PR TITLE
Changes to support ibmpowervs image deletion without cluster reference

### PR DESCRIPTION
**Pull request by Karthik-K-N**
**What this PR does / why we need it**:

This change removes the need of cluster reference while deleting the ibmpowervsimage resources

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
This is prerequisite for https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/899

/cc @mkumatag @Prajyot-Parab @dharaneeshvrd

File: controllers/ibmpowervsimage_controller.go [hunk 1]
```diff
@@ -97,8 +92,9 @@ func (r *IBMPowerVSImageReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return r.reconcileDelete(imageScope)
 	}
 
+	cluster, err := util.GetClusterByName(ctx, r.Client, ibmImage.Namespace, ibmImage.Spec.ClusterName)
 	if err != nil {
-		return reconcile.Result{}, errors.Errorf("failed to create scope: %+v", err)
+		return ctrl.Result{}, err
```
Comment (mkumatag) [inline controllers/ibmpowervsimage_controller.go:97]: any specific reason for using from the ctrl module? wondering if we can keep it sync with other returns in the program.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: controllers/ibmpowervsimage_controller.go (hunk 1 above)
Comment (Karthik-K-N) [reply controllers/ibmpowervsimage_controller.go:97]: No specific reason , I did not notice, it is being [there](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/controllers/ibmpowervsimage_controller.go#L68-L75) already, May be I will update to use `reconcile.Result{}`

Comment (Karthik-K-N) [reply controllers/ibmpowervsimage_controller.go:96]: Just updated one occurance of reconcile.Result{} to ctrl.Result{} to keep it consistent in the file

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**